modin/core/storage_formats/pandas/parsers.py

```diff
@@ -308,6 +308,10 @@ def parse(chunks, **kwargs):
         num_splits = kwargs.pop("num_splits", None)
         index_col = kwargs.get("index_col", None)
 
+        # `single_worker_read` just pass filename via chunks; need check
```

Comment:
this feels like a hack, could you please explain the call graph here? at least it contradicts the docstring of this function

Reply:
`CSVGlobDispatcher._read` -> `PandasCSVGlobParser.single_worker_read(inherited from PandasParser)` -> `PandasCSVGlobParser.parse`

Reply:
this still looks like very weird control flow, I'd like to refactor it somehow eventually (certainly in a separate PR)

Reply:
I agree, will put it in a separate issue.